etl/extractor/mainnet_impl.go

```diff
@@ -27,76 +35,165 @@ func NewMainNetExtractor(batchSize uint32, exporterClient exporter.RecordExporte
 	request := &exporter.GetRecords{Count: batchSize}
 	return &MainNetExtractor{
 		stopSignal:       make(chan bool, 1),
+		startStopMutes:   &sync.Mutex{},
 		client:           exporterClient,
 		request:          request,
 		mainJetDropsChan: make(chan *types.PlatformJetDrops),
 	}
 }
 
 func (m *MainNetExtractor) GetJetDrops(ctx context.Context) <-chan *types.PlatformJetDrops {
-	// from pulse, 0 means start to get from pulse number 0
-	//todo: add pulse fetcher
-	m.request.PulseNumber = 0
-	m.request.RecordNumber = 0
-	client := m.client
+	return m.mainJetDropsChan
+}
 
-	//todo: register event in some monitoring service
-	errorChan := make(chan error)
+func (m *MainNetExtractor) LoadJetDrops(ctx context.Context, fromPulseNumber int, toPulseNumber int) error {
+	if fromPulseNumber < 0 {
+		return errors.New("fromPulseNumber cannot be negative")
+	}
+	if toPulseNumber < 1 {
+		return errors.New("toPulseNumber cannot be less than 1")
+	}
+	if fromPulseNumber > toPulseNumber {
+		return errors.New("fromPulseNumber cannot be greater than toPulseNumber")
+	}
 
-	go func() {
-		//todo: enable logger
+	request := &exporter.GetRecords{
+		Count:        m.request.Count,
+		PulseNumber:  insolar.PulseNumber(fromPulseNumber),
+		RecordNumber: 0,
+	}
+	m.getJetDrops(ctx, request, fromPulseNumber, toPulseNumber, true)
+	return nil
+}
 
-		// logger := belogger.FromContext(ctx)
+func (m *MainNetExtractor) getJetDrops(ctx context.Context, request *exporter.GetRecords, fromPulseNumber int, toPulseNumber int, shouldReload bool) {
```

Comment:
multiple calls of this function may cause race on  m.needStop() function
no need fix for now but take care of this in future